### PR TITLE
Handle new gtfs-lib response for encoded polylines

### LINKED ISSUE #627: Editor: Show routes on a map causes out of memory issues
## Observed behavior (please include a screenshot if possible)

In the editor, turning on the pattern shapes overlay will cause out of memory issues for very large feeds.

## Expected behavior

There is currently no back end solution to prevent this, so this feature ought to be disabled for now.

## Steps to reproduce the problem

1. Open the editor for a very large feed
2. In the top left corner of the map check the 'Show routes on map' overlay
3. A massive GraphQL query requests the shape points for every pattern and kills the application

## Any special notes on configuration used

N/A

## Version of datatools-ui and datatools-server if applicable (exact commit hash or branch name)

latest dev

### PR DESCRIPTION
**Pull request by landonreed**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JSDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

fix #627. Alternative approach to #628. This works with changes from https://github.com/conveyal/gtfs-lib/pull/297

### CODE REVIEW

**Review by binh-dam-ibigroup — COMMENTED** [review]
Please address merge conflicts and build failures...

File: lib/editor/actions/tripPattern.js [hunk 1]
```diff
@@ -118,8 +119,25 @@ export function togglePatternEditing () {
...
+    shapes (limit: -1) {
+      shape_id
+      polyline
+    }
+  }
+}
+`
+    const variables = {namespace}
+    return dispatch(fetchGraphQL({query, variables}))
+      .then(data => dispatch(receiveGTFSEntities({namespace, component: 'pattern', data, editor: true, replaceNew: true})))
```
Comment (evansiroky) [inline lib/editor/actions/tripPattern.js:140]: Line too long.

**Review by evansiroky — CHANGES_REQUESTED** [review]
- Fix failing travis build
- Fix merge conflicts

File: lib/editor/actions/tripPattern.js [hunk 2]
```diff
@@ -118,8 +120,39 @@ export function togglePatternEditing () {
...
     const namespace = getEditorNamespace(feedId, getState())
-    return dispatch(fetchGTFSEntities({namespace, type: 'pattern', editor: true}))
+    const query = `
+query shapesQuery ($namespace: String) {
+  feed(namespace: $namespace) {
+    feed_version
+    feed_info {
+      feed_id
+    }
+    shapes_as_polylines (limit: -1) {
```
Comment (evansiroky) [inline lib/editor/actions/tripPattern.js:132]: Is `limit` needed here?

**Review by evansiroky — APPROVED** [review]
Approved, pending the changes in https://github.com/conveyal/gtfs-lib/pull/297 percolating it's way through the pipeline.

File: lib/editor/actions/tripPattern.js [hunk 3]
```diff
@@ -5,10 +5,11 @@ import {ActionCreators} from 'redux-undo'
 import {toast} from 'react-toastify'
 
 import {resetActiveGtfsEntity, savedGtfsEntity, updateActiveGtfsEntity, updateEditSetting} from './active'
-import {createVoidPayloadAction, secureFetch} from '../../common/actions'
+import {createVoidPayloadAction, fetchGraphQL, secureFetch} from '../../common/actions'
 import {snakeCaseKeys} from '../../common/util/map-keys'
 import {generateUID} from '../../common/util/util'
-import {fetchGTFSEntities} from '../../manager/actions/versions'
+import {showEditorModal} from './editor'
```
Comment (binh-dam-ibigroup) [inline lib/editor/actions/tripPattern.js:11]: abc order

**Review by binh-dam-ibigroup — APPROVED** [review]
Looks good and seems to be working after regenerating snapshots.

**Comment by landonreed** [comment]
Marked as blocked until datatools-server has the latest gtfs-lib release.

**Comment by evansiroky** [comment]
Something I just noticed... the Feed Version map still uses the raw GTFS shapes attributes to find the shapes: https://github.com/ibi-group/datatools-ui/blob/93a7dec276d2a2598a1197f468893f0c24d7fd0f/lib/gtfs/actions/shapes.js#L24-L48

Can this be updated too?

**Pull request merged**